### ISSUE TITLE
Run invoices and expenses claims through this repo?

### ISSUE THREAD

**Issue by sofer** [foundersandcoders/london-money#46]
I am not sure if this is a sensible suggestion, but does anyone else think it would be sensible to run contractor invoices and expense claims through GH issues, in this repo, or if people prefer, in a separate repo specifically for the purpose). The main advantage of doing it this way is that I, or whoever is managing our payments, can see incoming payments due and will not miss (as I am sad to say often happens) incoming emailed invoices and expense claims. The other advantage is transparency, not just about who is being paid what, but also about how efficiently we, as an organisation, are processing payments (and we are--I am--not doing so terribly efficiently right now).

@iteles @rebeccaradding @des-des @eliascodes @jsms90 @mattlub @m4v15 @JWLD @shiryz and everyone else who has been kept waiting recently and who might be waiting now.

**Comment by eliascodes** [comment]
I'm happy to go with whatever solution results in less work for the person managing payments. Does this mean we would be retiring Xero? Also how does this relate to #30?

**Comment by markwilliamfirth** [comment]
Probs not a good idea - contractor invoices etc can sometimes contain sensitive information

To achieve the same effect without making this info public you could just give everyone read access and ability to submit expenses through Xero

**Comment by eliascodes** [comment]
Yeah good point, also comments in issues are editable by anyone with little to no audit trail.

**Comment by sofer** [comment]
@eliascodes this could potentially supercede #30. Some of us have been using this workflow with limited success (it is essentially what @markwilliamfirth is suggesting above), but it is flawed in several ways. See https://github.com/foundersandcoders/london-money/issues/30#issuecomment-324889675

@eliascodes audit trail: I wasn't suggesting replacing Xero, just supplementing it with a tool to make it a bit easier to keep track of incoming payments due.

If people felt that their daily rate was sensitive information (presumably that is what @markwilliamfirth is talking about) then they could have the option to raise an issue without the totals.

**Comment by markwilliamfirth** [comment]
@sofer sensitive information from expenses / invoices can sometimes include (but not limited to):

- full names
- bank details
- company information
- rates
- addresses
- information about what places and what times people are purchasing things
- may show information indicating a client has procured something which the client may not want to be publicly known

**Comment by sohilpandya** [comment]
@sofer I like the idea of keeping things transparent. 

Also agree that Bank details, address and client details should be kept private, but name, company and rates should be open for everyone else to know.

**Comment by sofer** [comment]
The way I see this possibly working is that we create a new private repo called london-invoices to track the progress of invoice raising. The worksflow might look something like:

- Raise issue: "Invoice [#x] raised in Xero", add flag "awaiting approval" and assign to whoever is doing the book keeping;
- Add flag: "sent to client" once it has been approved and sent;
- Add flag: "client paid" once the client has paid;
- Add flag: "paid" once the contractors have all been paid;
- close issue

I will almost certainly find it more helpful to track this sort of thing in GitHub, which I am regularly checking, than in yet another spreadsheet, which I am likely to ignore, or even in Xero, which I only check periodically and sometimes still miss things like invoices awaiting approval. We can have a private repo called something london-invoices.

I am agnostic about putting people's rates into these issues. I like the idea of people's freelance rates being something that is open to the wider membership for scrutiny, but I do understand that some people may find this intrusive.

**Comment by markwilliamfirth** [comment]
@sofer all of those above functions are already in Xero so by putting this on GitHub it's doubling the workflow - if checking Xero is an issue perhaps a calendar notification / reminder would be a good solution?

**Comment by sofer** [comment]
@markwilliamfirth I can see you are not entirely keen on this idea, but Xero does not fit my use case, which is allowing everyone (not just Xero users) to keep track of our invoicing workflow. GitHub may not be the best tool for this and we may end up using something else, but it has the advantage of being used by everyone here every day.
